### PR TITLE
[1.0] Remove args from ray start

### PR DESCRIPTION
**Pull request by wuisawesome**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by ericl — APPROVED** [review]

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -341,66 +305,69 @@ def dashboard(cluster_config_file, cluster_name, port, remote_port):
     help="manually specify the socket path of the raylet process")
 @click.option(
     "--temp-dir",
+    hidden=True,
     default=None,
     help="manually specify the root temporary dir of the Ray process")
-@click.option(
```
Comment (rkooo567) [inline python/ray/scripts/scripts.py:346]: Is it okay to delete this?

Comment (wuisawesome) [reply python/ray/scripts/scripts.py:346]: According to this: https://docs.google.com/spreadsheets/d/1F6VZ1R_fgDzpRZjXI5_6gaUwW2CQKi431dCDTR-Vm8s/edit#gid=0&range=F12 yes

@raulchen maybe you can confirm?

Comment (raulchen) [reply python/ray/scripts/scripts.py:346]: @wuisawesome can you leave this option alone in this PR? We have another PR to remove this option. Removing it requires more code changes. https://github.com/ray-project/ray/pull/10594/

**Comment by ericl** [comment]
Breaks https://travis-ci.com/github/ray-project/ray/jobs/384279071

**Comment by ericl** [comment]
@wuisawesome please be more careful with the tests-ok tag, I did a double check and there were some failing tests!

File: java/test.sh [hunk 2]
```diff
@@ -56,8 +56,8 @@ case "${OSTYPE}" in
   darwin*) ip=$(ipconfig getifaddr en0);;
   *) echo "Can't get ip address for ${OSTYPE}"; exit 1;;
 esac
-RAY_BACKEND_LOG_LEVEL=debug ray start --head --redis-port=6379 --redis-password=123456 --code-search-path="$PWD/bazel-bin/java/all_tests_deploy.jar"
-RAY_BACKEND_LOG_LEVEL=debug java -cp bazel-bin/java/all_tests_deploy.jar -Dray.redis.address="$ip:6379"\
+RAY_BACKEND_LOG_LEVEL=debug ray start --head --port=6379 --redis-password=123456 --code-search-path="$PWD/bazel-bin/java/all_tests_deploy.jar"
+RAY_BACKEND_LOG_LEVEL=debug java -cp bazel-bin/java/all_tests_deploy.jar -Dray.address="$ip:6379"\
```
Comment (raulchen) [inline java/test.sh:60]: this line needs to be reverted (just `ray.redis.address`). I'm renaming this Java config in another PR: https://github.com/ray-project/ray/pull/10772/

**Comment by richardliaw** [comment]
Tests are failing: https://travis-ci.com/github/ray-project/ray/jobs/385303763

```
=================================== FAILURES ===================================
_________________________ test_calling_start_ray_head __________________________
call_ray_stop_only = None
    def test_calling_start_ray_head(call_ray_stop_only):
        # Test that we can call ray start with various command line
        # parameters. TODO(rkn): This test only tests the --head code path. We
        # should also test the non-head node code path.
    
        # Test starting Ray with no arguments.
        check_call_ray(["start", "--head"])
        check_call_ray(["stop"])
    
        # Test starting Ray with a redis port specified.
        check_call_ray(["start", "--head"])
        check_call_ray(["stop"])
    
        # Test starting Ray with a node IP address specified.
        check_call_ray(["start", "--head", "--node-ip-address", "127.0.0.1"])
        check_call_ray(["stop"])
    
        # Test starting Ray with a system config parameter set.
        check_call_ray([
            "start", "--head", "--system-config",
            "{\"metrics_report_interval_ms\":100}"
        ])
        check_call_ray(["stop"])
    
        # Test starting Ray with the object manager and node manager ports
        # specified.
        check_call_ray([
            "start", "--head", "--object-manager-port", "12345",
            "--node-manager-port", "54321"
        ])
        check_call_ray(["stop"])
    
        # Test starting Ray with the worker port range specified.
        check_call_ray([
            "start", "--head", "--min-worker-port", "50000", "--max-worker-port",
            "51000"
        ])
        check_call_ray(["stop"])
    
        # Test starting Ray with the number of CPUs specified.
        check_call_ray(["start", "--head", "--num-cpus", "2"])
        check_call_ray(["stop"])
    
        # Test starting Ray with the number of GPUs specified.
        check_call_ray(["start", "--head", "--num-gpus", "100"])
        check_call_ray(["stop"])
    
        # Test starting Ray with the max redis clients specified.
>       check_call_ray(["start", "--head", "--redis-max-clients", "100"])
/Users/travis/build/ray-project/ray/python/ray/tests/test_multi_node.py:419: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = ['start', '--head', '--redis-max-clients', '100'], capture_stdout = False
capture_stderr = False
    def check_call_ray(args, capture_stdout=False, capture_stderr=False):
        # We use this function instead of calling the "ray" command to work around
        # some deadlocks that occur when piping ray's output on Windows
        argv = ["ray"] + args
        if sys.platform == "win32":
            result = check_call_module(
                ray_main,
                argv,
                capture_stdout=capture_stdout,
                capture_stderr=capture_stderr)
        else:
            stdout_redir = None
            stderr_redir = None
            if capture_stdout:
                stdout_redir = subprocess.PIPE
            if capture_stderr and capture_stdout:
                stderr_redir = subprocess.STDOUT
            elif capture_stderr:
                stderr_redir = subprocess.PIPE
            proc = subprocess.Popen(argv, stdout=stdout_redir, stderr=stderr_redir)
            (stdout, stderr) = proc.communicate()
            if proc.returncode:
                raise subprocess.CalledProcessError(proc.returncode, argv, stdout,
>                                                   stderr)
E               subprocess.CalledProcessError: Command '['ray', 'start', '--head', '--redis-max-clients', '100']' returned non-zero exit status 2.

```

**Comment by ericl** [comment]
LINT, etc still failing. Please do not add tests ok until you have carefully checked the tests, thanks.

File: python/ray/scripts/scripts.py [hunk 3]
```diff
@@ -341,121 +305,80 @@ def dashboard(cluster_config_file, cluster_name, port, remote_port):
...
     "--java-worker-options",
     required=False,
+    hidden=True,
     default=None,
     type=str,
     help="Overwrite the options to start Java workers.")
 @click.option(
     "--code-search-path",
     default=None,
+    hidden=True,
```
Comment (kfstorm) [inline python/ray/scripts/scripts.py:321]: This option is now documented. We should not hide it, right?

Comment (wuisawesome) [reply python/ray/scripts/scripts.py:321]: thanks

**Pull request merged**